backend/Dockerfiles/Dockerfile.node14

```diff
@@ -28,7 +28,8 @@ RUN apt-get update && \
     rm -rf /var/lib/apt/lists/* && \
     ln -s /usr/bin/python3 /usr/bin/python && \
     npm install eslint-plugin-security && \
-    npm install -g eslint npm@7.8.0 && \
+    npm install typescript@5.1.6 @typescript-eslint/parser@5.50.0 && \
+    npm install -g eslint@8.57.0 npm@7.8.0 && \
```

Comment:
do we want to update the NPM version here while we are at it or is there a reason why we are keeping it at that version. Not sure if there would even be a benefit to updating npm here but wanted to open a potential discussion about it

Reply:
Newer versions of npm are not compatible with Node v14

In this release chart, it shows that Node v14.21.3 is released with npm v6.14.18. https://nodejs.org/en/about/previous-releases.

npm 7.8.0 is the latest version we can update to without any issues in Node 14

Reply:
gotcha sounds good thank you!